plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx

```diff
@@ -91,8 +131,29 @@ const ProductCollectionContent = (
 		return null;
 	}
 
+	// Let's not render anything until default attributes are set.
+	if ( ! isInitialAttributesSet.current ) {
+		return null;
+	}
+
 	return (
 		<div { ...blockProps }>
+			{ attributes.__privatePreviewState?.isPreview &&
+				isSelectedOrInnerBlockSelected && (
+					<Button
```

Comment:
Nit: I can see two issues with this and both are related to the UX:
- The button is clickable, while it shouldn't.
- Once clicked, it brings this outline.
- The default font family is the theme's one; we need to use the Gutenberg's admin one. 

That said, this UI feels "hacky" and I think we have two options:
1. Handle the button so it behaves as a label
2. Use a different component. 

WDYT?

Reply:
Forgot to add the screenshot: 
![Screenshot 2024-05-10 at 11 55 33 AM](https://github.com/woocommerce/woocommerce/assets/1105590/6a551a6d-47c7-4dd8-adbf-5e9c79a91677)

Reply:
I agree! For the first iteration, I didn't want to create a custom component, so I went with the one provided by Gutenberg. But I agree that this isn't great UX. IMO It makes sense to keep using Gutenberg component if we can. I added some custom CSS to hide outline & shadow in [6c6181f](https://github.com/woocommerce/woocommerce/pull/46369/commits/6c6181f340d51aa76b328570567816ad3f5d9095). What do you think about using custom CSS? I think, that's a good path for our case.
**Question:** Do you know any other component that we can use from Gutenberg? (In case I am not aware of it 🙂)

> The default font family is the theme's one; we need to use the Gutenberg's admin one.

Currently, we are using "inherit" for font-family. I can add CSS and provide specific font-family value which is used by GB admin i.e. `-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif`, but I don't feel that's right way. I think it's okay to keep using "inherit" value. In places like Inspector controls, "inherit" corresponds to admin font family. But in main content where blocks are added, it inherits from theme. IMO It's okay to keep using "inherit", otherwise Preview label will look different from other main content. What do you think?

Reply:
> I added some custom CSS to hide outline & shadow in [6c6181f](https://github.com/woocommerce/woocommerce/pull/46369/commits/6c6181f340d51aa76b328570567816ad3f5d9095).

Yup, that's much better! 

1. 
> otherwise Preview label will look different from other main content. What do you think?

I think this preview label should have the same font as the `Choose collection` within the toolbar. For example, setting the font like `var(--wp--preset--font-family--system-sans-serif)`

2.
Flagging (a rare case) that showing the label at all times when a child block is selected can create weird placements like this:
![ypReSJ.png](https://github.com/woocommerce/woocommerce/assets/1105590/fc23d17a-63e3-41fd-b4f2-7d76a1643fa7)

But... I'm seeing some significant changes on i2 designs for this API, which really affect the priority of these little fixes. So, I'm leaving these two up to you!

Reply:
> I think this preview label should have the same font as the `Choose collection` within the toolbar. For example, setting the font like `var(--wp--preset--font-family--system-sans-serif)`

Made the changes in [d530f00](https://github.com/woocommerce/woocommerce/pull/46369/commits/d530f008d09a1b2b9308738776af6a09178bd932) 🤝

> Flagging (a rare case) that showing the label at all times when a child block is selected can create weird placements like this:

Yeah. I don't think there is a good alternative solution to this. Let's leave it as it is for now. We will figure out a way in i2 🤝